server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.SetOnce;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.ClusterChangedEvent;
+import org.elasticsearch.cluster.ClusterStateListener;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.component.Lifecycle;
+import org.elasticsearch.common.logging.ESLogMessage;
+import org.elasticsearch.common.scheduler.SchedulerEngine;
+import org.elasticsearch.common.scheduler.TimeValueSchedule;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.gateway.GatewayService;
+import org.elasticsearch.health.node.selection.HealthNode;
+
+import java.io.Closeable;
+import java.time.Clock;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This class periodically logs the results of the Health API to the standard Elasticsearch server log file.
+ */
+public class HealthPeriodicLogger implements ClusterStateListener, Closeable, SchedulerEngine.Listener {
+    public static final String HEALTH_FIELD_PREFIX = "elasticsearch.health";
+
+    /**
+     * Creates a new HealthPeriodicLoggerResult.
+     * This creates a record that's used as the output of this feature.
+     *
+     * @param indicatorResults the results of the Health API call that will be used as the output.
+     */
+    public record HealthPeriodicLoggerResult(List<HealthIndicatorResult> indicatorResults) {
+
+        /**
+         * Create a Map of the results, which is then turned into JSON for logging
+         */
+        public Map<String, Object> toMap() {
```

Comment:
I think we should give a bit more context here. I had to look around to collect more info about this method.

My concerns:

- A method called `toMap` makes me think that this will return the indicator results grouped by something. But this method actually returns indicator name -> status.
- The return type `Map<String, Object>` has to be like this in order to be used by the `ESLogMessage`, but as a result we lose more information.

I would suggest to:

- Change the name of the method to something more descriptive, something like `toJsonFields` or `toLoggedFields` or something better.
- add to the javadoc an example of how the fields will look like or describe it but be specific that this contains only the indicator name and their status.

Do these suggestions make sense to you?

Reply:
Does make sense, good suggestion.